common/namespace/cache.go

```diff
@@ -684,9 +661,16 @@ func (entry *CacheEntry) GetConfig() *persistencespb.NamespaceConfig {
 	return entry.config
 }
 
-// GetReplicationConfig return the namespace replication config
-func (entry *CacheEntry) GetReplicationConfig() *persistencespb.NamespaceReplicationConfig {
-	return entry.replicationConfig
+// ActiveClusterName observes the name of the cluster that is currently active
+// for this namspace.
+func (entry *CacheEntry) ActiveClusterName() string {
+	return entry.replicationConfig.ActiveClusterName
+}
+
+// ClusterNames observes the names of the clusters to which this namespace is
+// replicated.
+func (entry *CacheEntry) ClusterNames() []string {
+	return append(([]string)(nil), entry.replicationConfig.Clusters...)
```

Comment:
Why not just:
```suggestion
	return entry.replicationConfig.Clusters
```
?

Reply:
Don't want to make the underlying entry mutable. Exposing the slice directly allows for mutation.

Reply:
Like:
```
entry.ClusterNames[1]="my new cluster name"
```
Who would do that in a sound mind? Anyway, I would use `copy` for this.

Reply:
I agree that it would be dumb code for someone to write. But if I'm going to say that something is immutable (which I intend to do), then I don't really want to leave it _mostly_ immutable.

I thought this particular form of copy was ok just because it's at the top of the SliceTricks wiki that we link to in Notion. But if people find it obscure I can do it with `copy`

Reply:
I think using append is fine. Maybe a comment to explain why not directly return entry.replicationConfig.Clusters.

Reply:
Too late. It's `copy` now. But I added a comment.